R/qc_ids.R

```diff
@@ -35,14 +35,14 @@
 #' )
 #' }
 qc_ids <-
-  function(data, sample, grouping, intensity, remove_na_intensities = TRUE, condition = NULL, title = "ID count per sample", plot = TRUE, interactive = TRUE) {
+  function(data, sample, grouping, intensity = NULL, remove_na_intensities = TRUE, condition = NULL, title = "ID count per sample", plot = TRUE, interactive = TRUE) {
     protti_colors <- "placeholder" # assign a placeholder to prevent a missing global variable warning
     utils::data("protti_colors", envir=environment()) # then overwrite it with real data
     if(remove_na_intensities == TRUE){
-      data <- data %>% 
+      data <- data %>%
+        dplyr::distinct({{sample}}, {{grouping}}, {{condition}}, {{intensity}}) %>%
```

Comment:
This line does not seem to be necessary. 
```suggestion
        dplyr::distinct({{sample}}, {{grouping}}, {{condition}}, {{intensity}}) %>%
```
usually a specific `drop_na({{ intensity }})` should be enough. This should not result in an empty data frame if there are intensities that are not NA. I think the problem was that the function was used wrong. So we should include an error if the user does not provide intensity.

Reply:
It is only required if remove_na_intensities = TRUE; I would not require the intensities in any case. 

I don't know, in this case, I think the function should be "idiot" proof, so that it also works for different inputs.